File: Number_of_Good_Pairs.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        pairs = {}
        count = 0

        for i in range(len(nums)):
            if nums[i] in pairs:
                count += pairs[nums[i]]
            
            pairs[nums[i]] = pairs.get(nums[i], 0) + 1
        
        return count

File: Number_of_Good_Pairs.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        Map<Integer, Integer> pairs = new HashMap<>();
        int count = 0;

        for (int i = 0; i < nums.length; i++) {
            if (pairs.containsKey(nums[i])) {
                count += pairs.get(nums[i]);
            }
            pairs.put(nums[i], pairs.getOrDefault(nums[i], 0) + 1);
        }

        return count;        
    }
}
